### PR TITLE
ARROW-479: Python: Test for expected schema in Pandas conversion

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: python/src/pyarrow/helpers.h [hunk 1]
```diff
@@ -31,6 +31,9 @@ using arrow::Type;
 PYARROW_EXPORT
 std::shared_ptr<DataType> GetPrimitiveType(Type::type type);
 
+PYARROW_EXPORT
+std::shared_ptr<DataType> GetTimestampType(arrow::TimeUnit unit);
```
Comment (wesm) [inline python/src/pyarrow/helpers.h]: You can use `::arrow::timestamp` here (see `arrow/type.h`)

**Review by wesm — COMMENTED** [review]
Definitely good to make the comparisons more precise. Minor comment, then can merge

File: python/src/pyarrow/helpers.h (hunk 1 above)
Comment (xhochy) [inline python/src/pyarrow/helpers.h]: Yes, there is actually no need for the `helpers.cc/h` except that I failed to include it correctly in `pyarrow/includes/libarrow.pxd`.

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**